cpp/include/cudf/detail/utilities/pinned_host_vector.hpp

```diff
@@ -169,7 +169,12 @@ class pinned_allocator {
    *        It is the responsibility of the caller to destroy
    *        the objects stored at \p p.
    */
-  __host__ inline void deallocate(pointer p, size_type /*cnt*/) { CUDF_CUDA_TRY(cudaFreeHost(p)); }
+  __host__ inline void deallocate(pointer p, size_type /*cnt*/)
+  {
+    auto dealloc_worked = cudaFreeHost(p);
+    (void)dealloc_worked;
```

Comment:
I have never seen a type cast without assignment. What does this line do?

Reply:
Generally, in Release mode, I believe this will suppress a variable-is-set-but-never-used warning from the compiler since the assert is compiled out.

Reply:
Correct on all accounts

Reply:
And where does the `status__` variable come from, seeing that it's never declared?

Reply:
Thanks @divyegala, that is a typo. I pushed a correction